### ISSUE TITLE
Check if `league/flysystem-memory` can be replaced by `mikey179/vfsstream`

### ISSUE THREAD

**Issue by kdambekalns** [neos/neos-development-collection#4711]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The former was added, evem though the latter was already a dependency. Find out why/if it's really needed.

### Expected Behavior

Only one dependency to do a thing is required.

### Environment

```markdown
- Neos: 9.0
```

**Comment by bwaidelich** [comment]
As written on [Slack](https://neos-project.slack.com/archives/C3MCBK6S2/p1699369742774429?thread_ts=1699368079.907069&cid=C3MCBK6S2):

> league/flysystem is only a dependency of the Neos.ContentRepository.Export package. It's the most popular abstraction
> for file systems and our reasoning for adding it was:
> * It makes testing much more easy and reliable
> * It allows for replacing the adapter without inventing yet another abstraction wheel (e.g. use league/flysystem-ziparchive to create zip-exports, export to cloud, ...)

**Comment by mhsdesign** [comment]
Yes I’m fine with this new abstraction and added the memory adapter to the dev collection (used as dev dependency for testing)
